lib/users.js

```diff
@@ -116,23 +117,118 @@ Users.prototype.validate = function (userName, password, callback) {
   });
 };
 
+
+/**
+ * Change password of user,
+ * And returns the profile or one of these errors:
+ *
+ * -  WrongPassword                   { profile object }
+ * -  WrongUsername                   { username string }
+ * -  UnexpectedError                 { inner Error }
+ * -  InsufficientAccessRightsError   { inner Error }
+ *
+ * @param  {[type]}   userName [description]
+ * @param  {[type]}   password [description]
+ * @param  {Function} callback [description]
+ */
+Users.prototype.changePassword = function (userName, password, callback) {
+  var self = this;
+  var client = this._client;
+
+  var log = logger(userName);
+
+  self.getByUserName(userName, function (err, profile) {
+    if (err) {
+      return callback(UnexpectedError.wrap(err));
+    }
+
+    if (!profile) {
+      return callback(new WrongUsername(userName));
+    }
+
+    // AD will search and delay an error till later if no password is given
+    if (password === '') {
+      return callback(new WrongPassword(profile));
+    }
+
+    log('Change password for DN "' + profile.dn.green + '"');
+
+    var change = new ldap.Change({
+      operation: 'replace',
+      modification: {
+        userPassword: password
```

Comment:
Added the parameter `CHANGE_PASSWORD_ATTRIBUTE_NAME`, defaults to `userPassword `.